### ISSUE TITLE
Race condition: during the un-deployment, randomly the secret from the management namespace is removed before the services that were created against this secret. 

### ISSUE THREAD

**Issue by ptesny** [SAP/sap-btp-service-operator#447]
When trying to use the [subaccount-for-a-serviceinstance-resource](https://github.com/SAP/sap-btp-service-operator?tab=readme-ov-file#subaccount-for-a-serviceinstance-resource) with ArgoCDaaS I am observing the following race condition:
My secret is in the management namespace; and the BTP services are created in other business namespaces.
That works fine. 

However, during the un-deployment, randomly the secret from the management namespace is removed before the services that were created against this secret. 

In order to mitigate it I must manually add the secret to the management namespace. 
In other words the namespaced and subaccount-for-a-serviceinstance secrets cannot be managed by CI/CD tools
Is that by design?

**Comment by I065450** [comment]
Hi @ptesny 

This is the same issue as:
https://github.com/SAP/sap-btp-service-operator/issues/449

Regards,
Naama
